Automattic/vip-go-mu-plugins
# Adds Parsely configs in SDS and extended integration types with disabled options

## Description

Previously we only tracked basic Parse.ly attributes in SDS ( `active, integration_type, version` ) but now we also introduced an additional `configs` key which tracks more details about how Parse.ly plugin is integrated.

Example of possible config attributes:
```
{
        "is_pinned_version": false,
	"site_id": "example.com",
        "have_api_secret": true,
	"is_javascript_disabled": false,
	"is_autotracking_disabled": false,
	"should_track_logged_in_users": false,
	"tracked_post_types": [
		{ "type": "post", "track_type": "post" },
		{ "type": "page", "track_type": "non-post" },
		{ "type": "cpt_post", "track_type": "do-not-track" }
	]
}
```

Apart from this we have also extended the possible values in `integration_type` key and now also tracks disabled status of the plugin.

```
- DISABLED_MUPLUGINS_FILTER
- DISABLED_MUPLUGINS_SILENT_OPTION
```

## Changelog Description

Introduced `configs` key in `parsely` attributes which tracks more info about how the plugin is used and also extended possible integration types to also track the disable status of the plugin

## Checklist

Please make sure the items below have been covered before requesting a review:

- [x] This change has relevant tests (if applicable).

## Steps to Test

1. Pull branch

2. Run `vip dev-env start` (if dev env already exists else create it via help from readme)

3. Test `config` attributes

    3.1. Enable `wp-parsely` by following [VIP Docs](https://docs.wpvip.com/how-tos/enabling-and-disabling-parsely/)

    3.2. Pull specifc branch from SDS that have these config changes (if not merged), run `docker-compose up` and make changes in `put_site_details` function so that it talks to the SDS running locally
    
    3.3. For verification we can either use `print_r` in `put_site_details` function and verify the body of `PUT` request or we can also verify the values from locally available SDS db
    
    3.4. Makes changes in plugin settings by going into `WP Admin > Settings > Parsely` and verify that `parsely.configs` key inside `PUT /site-details/sites` request have the expected values in it
    
    3.5. For testing `is_pinned_version` key inside `configs` you will have to use `wpvip_parsely_version` filter in `plugin-loader.php`

4. Test new values of `integration_type` 

    4.1. Use `wpvip_parsely_load_mu` filter to disable the plugin and verify that `parsely.integration_type` inside `PUT /site-details/sites` request have value `DISABLED_MUPLUGINS_FILTER`
    
    4.2. Use `_wpvip_parsely_mu` option to disable the plugin and verify that `parsely.integration_type` inside `PUT /site-details/sites` request have value `DISABLED_MUPLUGINS_SILENT_OPTION`
